music/src/core/visualizer.ts

```diff
@@ -441,3 +441,1266 @@ export class PianoRollSVGVisualizer extends BaseVisualizer {
+    accidental: [ 3,  0,  0,  1,  0,  3,  0,  3,  0,  0,  1,  0] }, {
+    steps:      [ 0, -1, -1, -2, -2, -3, -4, -4, -5, -5, -6, -6], // Bb
+    accidental: [ 0,  2,  0,  0,  3,  0,  2,  0,  2,  0,  0,  3] }, {
+    steps:      [ 0,  0, -1, -1, -2, -3, -3, -4, -4, -5, -5, -6], // B
+    accidental: [ 3,  0,  3,  0,  0,  3,  0,  3,  0,  3,  0,  0] }
+];
+
+const KEY_ACCIDENTALS = [
+  {accidental: 1, pitches: []},                       // C
+  {accidental: 2, pitches: [70, 75, 68, 73, 66]},     // Db
+  {accidental: 1, pitches: [78, 73]},                 // D
+  {accidental: 2, pitches: [70, 75, 68]},             // Eb
+  {accidental: 1, pitches: [78, 73, 80, 75]},         // E
+  {accidental: 2, pitches: [70]},                     // F
+  {accidental: 2, pitches: [70, 75, 68, 73, 66, 71]}, // Gb
+  {accidental: 1, pitches: [78]},                     // G
+  {accidental: 2, pitches: [70, 75, 68, 73]},         // Ab
+  {accidental: 1, pitches: [78, 73, 80]},             // A
+  {accidental: 2, pitches: [70, 75]},                 // Bb
+  {accidental: 1, pitches: [78, 73, 80, 75, 70]}      // B
+];
+
+interface QNote {
+  start: number; // In quarter note quantities (float)
+  length: number; // In quarter note quantities (float)
+  vSteps: number; // In score steps (int, 2 per staff line), vertically invert.
+  accidental: number; // Kind: -1 = natural, 0 = none, 1 = accidental
+  opacity: number; // from 0.0 (fully transparent) to 1.0 (fully opaque)
+  pitch: number; // MIDI/Protobuf pitch needed to name SVG Group
+  xHeadRight: number; // x position at the right of the note head to draw ties
+  tiedFrom?: QNote; // Reference to previous tied note
+  g?: SVGElement; // SVG Group to hold tied notes into
+}
+
+interface BlockDetails {
+    maxVStep: number;
+    minVStep: number;
+    restToNextLength: number;
+    notes: QNote[];
+}
+
+/**
+ * Enumeration of different ways of horizontal score scrolling, like paginaged
+ * (PAGE is default value), note by note (NOTE) or in packed chunks by doing 
+ * scroll just on bar starting (BAR).
+ */
+export enum ScrollType {
+  PAGE = 0,
+  NOTE = 1,
+  BAR = 2
+}
+
+/**
+ * An interface for providing extra configurable properties to a Visualizer
+ * extending the basic configurable properties of `VisualizerConfig`.
+ * @param defaultKey The musical key the score must use to adapt the score to 
+ * the right accidentals. It can be overwritten with 
+ * `NoteSequence.keySignatures` value at time or step 0. If not assigned it 
+ * will be asumed C key.
+ * @param instruments The subset of the `NoteSequence` instrument track numbers 
+ * which should be merged and displayed. If not assigned or equal to [] it will 
+ * be used all instruments altogether.
+ * @param scrollType Sets scrolling to follow scoreplaying in different ways 
+ * according to `ScrollType` enum values.
+ */
+export interface AdvancedVisualizerConfig extends VisualizerConfig {
```

Comment:
You can just call this StaffVisualizerConfig, since it's not used by anything else

Reply:
Sure! It was the original name and I was planning to change it exactly as you mention, but I forgot it. Thanks.